pkg/k8s/scanner/scanner.go

```diff
@@ -431,3 +442,27 @@ func toProperties(props map[string]string, namespace string) []core.Property {
 	})
 	return properties
 }
+
+func generatePURL(name, ver, nodeName string) *purl.PackageURL {
+	// Identify k8s distribution. An empty namespace means upstream.
+	var namespace string
+	switch {
+	case strings.Contains(ver, "eks"):
+		namespace = purl.NamespaceEKS
+	case strings.Contains(ver, "gke"):
+		namespace = purl.NamespaceGKE
+	case strings.Contains(ver, "rke2"):
+		namespace = purl.NamespaceRKE
+	case strings.Contains(ver, "hotfix"):
+		if !strings.Contains(nodeName, "aks") {
+			// Unknown k8s distribution
+			return nil
+		}
+		namespace = purl.NamespaceAKS
+	case strings.Contains(nodeName, "ocp"):
+		namespace = purl.NamespaceOCP
+	}
+	return &purl.PackageURL{
+		PackageURL: *packageurl.NewPackageURL(purl.TypeK8s, namespace, name, ver, nil, ""),
```

Comment:
nit: note sure if for now , it will be better to use `option pattern` to avoid passing `nil` and `empty strings`